application-admintools-default/src/main/java/com/xwiki/admintools/internal/usage/AbstractInstanceUsageProvider.java

```diff
@@ -0,0 +1,191 @@
+                            return wikiData.getUserCount().equals(Long.parseLong(filter.getValue()));
+                        case ATTACHMENTS_SIZE_KEY:
+                            String[] interval = filter.getValue().split(INTERVAL_SEPARATOR);
+                            long attachmentsSize = wikiData.getAttachmentsSize();
+                            long lowerBound = Long.parseLong(interval[0]);
+                            long upperBound = "x".equals(interval[1]) ? Long.MAX_VALUE : Long.parseLong(interval[1]);
+                            return attachmentsSize > lowerBound && attachmentsSize < upperBound;
+                        case ATTACHMENTS_COUNT_KEY:
+                            return wikiData.getAttachmentsCount().equals(Long.parseLong(filter.getValue()));
+                        case DOCUMENTS_COUNT_KEY:
+                            return wikiData.getDocumentsCount().equals(Long.parseLong(filter.getValue()));
+                        case TOTAL_COUNT_KEY:
+                            return wikiData.getTotal().equals(Long.parseLong(filter.getValue()));
+                        default:
+                            throw new IllegalArgumentException("Invalid filter field: " + filter.getKey());
+                    }
+                });
+    }
+
+    /**
+     * Sort the given {@link List} over a given column and in a given order.
+     *
+     * @param list the {@link List} to be sorted.
+     * @param sortColumn the column after which to be sorted.
+     * @param order the sort oder.
+     */
+    public void applySort(List<WikiUsageResult> list, String sortColumn, String order)
+    {
+        Comparator<WikiUsageResult> comparator = null;
+        switch (sortColumn) {
+            case WIKI_NAME_KEY:
+                comparator = Comparator.comparing(WikiUsageResult::getWikiName);
+                break;
+            case USER_COUNT_KEY:
+                comparator = Comparator.comparing(WikiUsageResult::getUserCount);
+                break;
+            case ATTACHMENTS_SIZE_KEY:
+                comparator = Comparator.comparing(WikiUsageResult::getAttachmentsSize);
+                break;
+            case ATTACHMENTS_COUNT_KEY:
+                comparator = Comparator.comparing(WikiUsageResult::getAttachmentsCount);
+                break;
+            case DOCUMENTS_COUNT_KEY:
+                comparator = Comparator.comparing(WikiUsageResult::getDocumentsCount);
+                break;
+            case TOTAL_COUNT_KEY:
+                comparator = Comparator.comparing(WikiUsageResult::getTotal);
+                break;
+            default:
+                break;
+        }
+        if (comparator != null) {
+            if (DESCENDING_ORDER.equals(order)) {
+                comparator = comparator.reversed();
+            }
+            list.sort(comparator);
+        }
+    }
+
+    /**
+     * Sort the given {@link List} over a given column and in a given order.
```

Comment:
`Sort the given {@link List} of documents over a given column and in a given order` , so to differentiate this from the method above.
Also, this method is not really specific to spam pages and could be renamed to `applyDocumentsSort`, to not link it only to spam pages, you could need it in the future